lib/src/components/NxTabs/NxTabs.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, {Children, cloneElement, isValidElement, useMemo} from 'react';
+import classnames from 'classnames';
+
+import {Props as NxTabProps} from '../NxTab/types';
+import {Props as NxTabPanelProps} from '../NxTabPanel/types';
+
+import { Props, propTypes } from './types';
+export { Props } from './types';
+
+interface ActiveTabContextType {
+  activeTab: number | null | undefined;
+  onTabSelect: (index: number | null | undefined) => void;
+};
+
+export const ActiveTabContext = React.createContext<ActiveTabContextType>({activeTab: null, onTabSelect: () => {}});
+
+let tabId = 0;
+
+const NxTabs = function NxTabsElement(props: Props) {
+  const {activeTab, onTabSelect, id, className, children, ...attrs} = props;
+
+  const activeTabContext: ActiveTabContextType = {
+    activeTab,
+    onTabSelect: onTabSelect || (() => {})
+  };
+
+  const [tabList, ...tabPanels] = Children.toArray(children);
+
+  if (!isValidElement(tabList)) {
+    console.error('NxTabs must have an NxTabList');
+    return null;
+  }
+  else if (tabList.props.children.length === 0) {
+    console.error('NxTabs must have at least one NxTab');
+    return null;
+  }
+
+  const rootId = id || 'nx-tabs-' + useMemo(() => tabId++, []);
```

Comment:
This paragraph from the react docs causes me to have concerns about this:

> You may rely on useMemo as a performance optimization, not as a semantic guarantee. In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without useMemo — and then add it to optimize performance.

Reply:
> This paragraph from the react docs causes me to have concerns about this:

I'm wondering if we should just require the caller to specify an id

Reply:
I was worried about that initially as well, however I think this is fine. Even if the rootId gets recomputed, everything would still be rendered with a safe id.

The bigger concern I have with this implementation is more around doing enough renders that the tabId gets extremely large. What I'm trying to accomplish here is to ensure that each tab panel rendered on the screen has unique ids for the accessibility attributes. This should work for the most part, but maybe I could reset the tab id once it gets beyond some maximum value (ie we never expect more than 100 NxTab components on the screen at the same time, so once we hit that we reset it to 0 to prevent the value from becoming too large over time).